packages/scheduler/src/SchedulerPostTask.js

```diff
@@ -150,30 +151,28 @@ function runTask<T>(
           ),
           continuationOptions,
         )
-        .catch(handlePostTaskError);
+        .catch(handleAbortError);
     }
+  } catch (error) {
+    // We're inside a promise. If we don't handle this error, then it will
+    // trigger an "Unhandled promise rejection" error. We don't want that, but
+    // we do want the default error reporting behavior that normal (non-Promise)
+    // tasks get for unhandled errors.
+    //
+    // So we'll re-throw the error inside a regular browser task.
+    setTimeout(() => {
+      throw error;
+    });
   } finally {
     currentPriorityLevel_DEPRECATED = NormalPriority;
   }
 }
 
-function handlePostTaskError(error) {
-  // This error is either a user error thrown by a callback, or an AbortError
-  // as a result of a cancellation.
-  //
-  // User errors trigger a global `error` event even if we don't rethrow them.
-  // In fact, if we do rethrow them, they'll get reported to the console twice.
-  // I'm not entirely sure the current `postTask` spec makes sense here. If I
-  // catch a `postTask` call, it shouldn't trigger a global error.
-  //
+function handleAbortError(error) {
```

Comment:
Doesn't `handleAbortError` assume that the only way a postTask rejects is via an abort?

Reply:
Yeah. I'm pretty sure that's a correct assumption. Since the user code is wrapped in its own try/catch.